include/deal.II/base/symmetric_tensor.h

```diff
@@ -1584,10 +1590,37 @@ namespace internal
     };
 
 
+
+    /**
+     * Replacement of std::swap with device support
+     * for float and double (std::swap is host only)
+     */
+    template <class NumberType>
+    DEAL_II_HOST_DEVICE void
+    swap(NumberType &x, NumberType &y) noexcept
+    {
+      if constexpr (std::is_same_v<NumberType, double> ||
+                    std::is_same_v<NumberType, float> ||
+                    std::is_same_v<NumberType, unsigned int>)
+        {
+#  if DEAL_II_KOKKOS_VERSION_GTE(4, 3, 0)
+          Kokkos::kokkos_swap(x, y);
+#  else
+          NumberType tmp = x;
+          x              = y;
+          y              = tmp;
```

Comment:
I though I had suggested using `std::move()` for the right hand sides of these three lines. But I may not recall correctly. Could you try whether that works?

Reply:
It is only used on device for float, double, int. I don't think ``move`` matters here.

Reply:
Oh, right. Yes, no worries then.